lib/gpx/geojson.rb

```diff
@@ -0,0 +1,222 @@
+#--
+# Copyright (c) 2017  Nick Knipe
+#
+# Permission is hereby granted, free of charge, to any person obtaining
+# a copy of this software and associated documentation files (the
+# "Software"), to deal in the Software without restriction, including
+# without limitation the rights to use, copy, modify, merge, publish,
+# distribute, sublicense, and/or sell copies of the Software, and to
+# permit persons to whom the Software is furnished to do so, subject to
+# the following conditions:
+#
+# The above copyright notice and this permission notice shall be
+# included in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+# NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+# LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+# OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+# WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+#++
+require 'json'
+
+module GPX
+  # Class to parse GeoJSON LineStrings, MultiLineStrings, Points,
+  # and MultiPoint geometric objects to GPX format. For the full
+  # specification of GeoJSON, see:
+  #   http://geojson.org/geojson-spec.html
+  # Note that GeoJSON coordinates are specified in lon/lat format,
+  # instead of the more traditional lat/lon format.
+  #
+  class Geojson
+    class << self
+      FEATURE = 'Feature'.freeze
+      LINESTRING = 'LineString'.freeze
+      MULTILINESTRING = 'MultiLineString'.freeze
+      POINT = 'Point'.freeze
+      MULTIPOINT = 'MultiPoint'.freeze
+
+      # Conversion can be initiated by either specifying a file,
+      # file name, or by passing in GeoJSON data as a string.
+      # Examples:
+      #   GPX::Geojson.convert_to_gpx(geojson_file: 'mygeojsonfile.json')
+      # or
+      #   file = File.new('mygeojsonfile.json', 'r')
+      #   GPX::Geojson.convert_to_gpx(geojson_file: file)
+      # or
+      #   data = JSON.generate(my_geojson_hash)
+      #   GPX::Geojson.convert_to_gpx(geojson_data: data)
+      #
+      # Returns a GPX::GPX_File object populated with the converted data.
+      #
+      def convert_to_gpx(opts = {})
```

Comment:
On further reflection, is there an advantage to changing the method interface from class methods to instance methods? The functionality would of course be the same, but since we are not actually storing any data inside any given instance for manipulation -- it is just JSON in GPX out -- my sense is that a class method interface makes for a simpler API.

What do you think?